### PR TITLE
Fix crash in Implement-Abstract-Class when dealing with partial classes.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/13149

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Features/CSharp/Portable/CodeFixes/ImplementAbstractClass/ImplementAbstractClassCodeFixProvider.cs [hunk 1]
```diff
@@ -1,99 +1,21 @@
...
-using Microsoft.CodeAnalysis.ImplementAbstractClass;
-using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.ImplementAbstractClass
 {
     [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.ImplementAbstractClass), Shared]
     [ExtensionOrder(After = PredefinedCodeFixProviderNames.GenerateType)]
-    internal class ImplementAbstractClassCodeFixProvider : CodeFixProvider
+    internal class ImplementAbstractClassCodeFixProvider : 
+        AbstractImplementAbstractClassCodeFixProvider<ClassDeclarationSyntax>
```
Comment (Pilchie) [inline src/Features/CSharp/Portable/CodeFixes/ImplementAbstractClass/ImplementAbstractClassCodeFixProvider.cs]: `Abstract` much? 👍

File: src/Features/Core/Portable/ImplementAbstractClass/AbstractImplementAbstractClassService.cs [hunk 2]
```diff
@@ -7,31 +7,37 @@
...
         protected AbstractImplementAbstractClassService()
         {
         }
 
-        protected abstract bool TryInitializeState(Document document, SemanticModel model, SyntaxNode classNode, CancellationToken cancellationToken, out INamedTypeSymbol classType, out INamedTypeSymbol abstractClassType);
+        protected abstract bool TryInitializeState(Document document, SemanticModel model, TClassSyntax classNode, CancellationToken cancellationToken, out INamedTypeSymbol classType, out INamedTypeSymbol abstractClassType);
 
-        public Task<Document> ImplementAbstractClassAsync(Document document, SemanticModel model, SyntaxNode node, CancellationToken cancellationToken)
+        public async Task<Document> ImplementAbstractClassAsync(
+            Document document, SyntaxNode classNode, CancellationToken cancellationToken)
```
Comment (Pilchie) [inline src/Features/Core/Portable/ImplementAbstractClass/AbstractImplementAbstractClassService.cs]: Why not make this `TClassSyntax` instead of casting?

**Comment by Pilchie** [comment]
👍, does Implement Interface suffer from a similar problem?

**Comment by CyrusNajmabadi** [comment]
I checked ImplementInterface and it works very differently and is unaffected by this.

**Pull request merged**